File: javascript/etl/etl.js
var ETL = function(){};

ETL.prototype.transform = function(old) {
    newObj = {};

    Object.keys(old).forEach(function(key) {
        if (old[key] instanceof Array) {
            old[key].forEach(function(value) {
                addAt(newObj, value, key);
            });
        } else {
            addAt(newObj, old[key], key);
        }
    });
    
    return newObj;
};

function addAt(obj, key, value) {
    obj[newKey(key)] = Number(value);
}

function newKey(str) {
    return String(str).toLowerCase();
};

module.exports = ETL;

File: java/etl/src/main/java/Etl.java
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Etl {
	
	public Map<String, Integer> transform(Map<Integer, List<String>> old) {
		Map<String, Integer> result = new HashMap<>();
		
		for (Integer key : old.keySet()) {
			for (String letter : old.get(key)) {
				result.put(letter.toLowerCase(), key);
			}
		}
		
		return result;
	}

}